analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -113,20 +112,20 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
                 ? semanticModel.GetSymbolInfo(objectCreation.Type).Symbol
                 : semanticModel.GetSymbolInfo(constructorCall).Symbol?.ContainingType;
 
-        private static SyntaxNode GetAssignmentTargetVariable(SyntaxNode objectCreation) =>
+        private static ImmutableArray<SyntaxNode> GetAssignmentTargetVariables(SyntaxNode objectCreation) =>
             objectCreation.GetFirstNonParenthesizedParent() switch
             {
-                AssignmentExpressionSyntax assignment => assignment.Left,
-                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration => variableDeclaration.Variables.Last(),
-                _ => null
+                AssignmentExpressionSyntax assignment => assignment.AssignmentTargets(),
+                EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration =>
+                    variableDeclaration.Variables.Select(x => (SyntaxNode)x).ToImmutableArray(),
+                ArgumentSyntax argument when argument.Ancestors().OfType<AssignmentExpressionSyntax>().FirstOrDefault() is { } expressionSyntax =>
+                    expressionSyntax.AssignmentTargets(),
+                _ => ImmutableArray<SyntaxNode>.Empty
             };
 
-        private static IPropertySymbol GetPropertySymbol(AssignmentExpressionSyntax assignment, SemanticModel model) =>
-            model.GetSymbolInfo(assignment.Left).Symbol as IPropertySymbol;
-
-        private static ISymbol GetAccessedVariable(AssignmentExpressionSyntax assignment, SemanticModel model)
+        private static ISymbol GetAccessedVariable(SyntaxNode node, SemanticModel model)
```

Comment:
This method should use the same logic as GetAssignmentTargetVariables; let's review it once GetAssignmentTargetVariables is in a better state.